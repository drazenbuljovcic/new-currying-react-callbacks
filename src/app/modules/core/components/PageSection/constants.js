export const DEFAULT_COMPONENT_ID = 'PageSection';
export const DEFAULT_CONTAINER_COMPONENT_ID = 'PageSection-Container';
export const DEFAULT_ITEM_COMPONENT_ID = 'PageSection-Item';