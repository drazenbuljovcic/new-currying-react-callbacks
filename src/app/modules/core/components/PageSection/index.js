import React from 'react';
import PropTypes from 'prop-types';

import DefaultContainer from './Containers/Default';
import DefaultItem from './Items/Default';

const PageSection = ({ Container, Item }) => (
  <Container
    Item={Item} />
);

PageSection.propTypes = {
  Container: PropTypes.oneOfType([PropTypes.node, PropTypes.element, PropTypes.func]),
  Item: PropTypes.oneOfType([PropTypes.node, PropTypes.element, PropTypes.func]),
};

PageSection.defaultProps= {
  Container: DefaultContainer,
  Item: DefaultItem,
};

export default PageSection;
