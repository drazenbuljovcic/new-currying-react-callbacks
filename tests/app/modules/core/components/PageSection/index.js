import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PageSection from 'app/modules/core/components/PageSection';
import DefultPageSectionContainer from 'app/modules/core/components/PageSection/Components/Containers/Default';
import DefultPageSectionItem from 'app/modules/core/components/PageSection/Components/Items/Default';

const TestContainer = () => {};
const TestItem = () => {};

describe('Page section component', () => {
  it('should render with it\'s default snapshot', () => {
    const section = shallow(<PageSection />);
    expect(toJson(section)).toMatchSnapshot();
  });

  it('should render with the default container', () => {
    const section = shallow(<PageSection />);
    expect(section.find(DefultPageSectionContainer).length).toEqual(1);
  });

  it('should render with a custom container', () => {
    const section = shallow(<PageSection
      Container={TestContainer} />);
    expect(section.find(TestContainer).length).toEqual(1);
  });

  describe('Page section container component', () => {
    it('should render with the default item', () => {
      const container = shallow(<DefultPageSectionContainer items={[{ id: 'item 1' }]} />);
      expect(container.find(DefultPageSectionItem).length).toEqual(1);
    });

    it('should render with a custom item', () => {
      const container = shallow(<DefultPageSectionContainer
        Item={TestItem}
        items={[{ id: 'item 1' }]} />);
      expect(container.find(TestItem).length).toEqual(1);
    });
  });
});
