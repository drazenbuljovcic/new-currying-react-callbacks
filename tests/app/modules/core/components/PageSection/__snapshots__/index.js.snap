// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page section component should render with it's default snapshot 1`] = `
<styled.div
  className="PageSection null"
  id="PageSection-null"
  style={Object {}}
>
  <DefaultContainer
    Item={[Function]}
    ItemProps={
      Object {
        "componentId": "PageSection-Item",
      }
    }
    backgroundColor={null}
    className={null}
    componentId="PageSection-Container"
    id={null}
    items={Array []}
    style={Object {}}
  />
</styled.div>
`;
